cpmpy/solvers/pindakaas.py

```diff
@@ -0,0 +1,302 @@
+# /usr/bin/env python
+"""
+Interface to the Pindakaas solver's Python API.
+
+Pindakaas is an open-source Rust library for encoding propositional and pseudo-Boolean constraints to SAT, with support for incremental solving and assumptions.
+
+Always use :func:`cp.SolverLookup.get("pindakaas") <cpmpy.solvers.utils.SolverLookup.get>` to instantiate the solver object.
+
+============
+Installation
+============
+
+Requires that the 'pindakaas' optional dependency is installed:
+
+.. code-block:: console
+
+    $ pip install pindakaas
+
+Detailed installation instructions available at:
+
+- https://pypi.org/project/pindakaas/
+- https://github.com/pindakaashq/pindakaas
+
+The rest of this documentation is for advanced users.
+
+===============
+List of classes
+===============
+
+.. autosummary::
+    :nosignatures:
+
+    CPM_pindakaas
+
+==============
+Module details
+==============
+"""
+
+import importlib
+import time
+from datetime import timedelta
+
+from ..exceptions import NotSupportedError
+from ..expressions.utils import eval_comparison
+from ..expressions.core import BoolVal, Comparison
+from ..expressions.variables import NegBoolView, _BoolVarImpl, _IntVarImpl
+from ..transformations.decompose_global import decompose_in_tree
+from ..transformations.flatten_model import flatten_constraint
+from ..transformations.get_variables import get_variables
+from ..transformations.int2bool import _decide_encoding, _encode_int_var, int2bool
+from ..transformations.linearize import linearize_constraint
+from ..transformations.normalize import simplify_boolean, toplevel_list
+from ..transformations.reification import only_bv_reifies, only_implies
+from ..transformations.safening import no_partial_functions
+from .solver_interface import ExitStatus, SolverInterface
+
+
+class CPM_pindakaas(SolverInterface):
+    """
+    Interface to Pindakaas' Python API.
+
+    Creates the following attributes (see parent constructor for more):
+
+    - ``pdk_solver``: the `pindakaas` solver or formula object
```

Comment:
please document all of 'ivarmap', 'encoding', 'unsatisfiable', 'core' here; (and none in the init below)